src/constraint/encoding.ts

```diff
@@ -31,7 +31,16 @@ export class EncodingConstraintModel extends AbstractConstraintModel {
       // TODO: extract as a method and do unit test
       const hasRequiredPropertyAsEnumSpec = some(
         this.constraint.properties,
-        (prop) => isEnumSpec(encQ[prop])
+        function(prop) {
+          if (hasNestedProperty(prop)) {
+            let nestedEncProp = getNestedEncodingProperty(prop);
+            let parent = nestedEncProp.parent;
+            let child = nestedEncProp.child;
+            return isEnumSpec(encQ[parent][child]);
+          } else {
```

Comment:
You `if` always return.  So you can actually remove `else` to have less text.  
(just a style comment -- no big deal.  But I prefer that you remove :) )